deduped-dashmap/src/lib.rs

```diff
@@ -98,15 +122,15 @@ where
     /// keys refer to the storage item, it will also be removed.
     pub fn remove(&self, key: K) {
         let key_desc = format!("{:?}", key);
-        if let Entry::Occupied(mut occupied_pointer_entry) = self.pointers.entry(key) {
-            let pointer = occupied_pointer_entry.get_mut();
+        if let Entry::Occupied(occupied_pointer_entry) = self.pointers.entry(key) {
+            let pointer = occupied_pointer_entry.remove();
             match self.storage.entry(pointer.hash) {
                 Entry::Occupied(mut occupied_storage_entry) => {
                     let item = occupied_storage_entry.get_mut();
                     if item.refcount > 1 {
                         item.refcount -= 1;
                     } else {
-                        occupied_pointer_entry.remove();
+                        occupied_storage_entry.remove();
```

Comment:
Ah, nice catch. The pointer is not removed if the value's refcount is greater than 1. Otherwise, the value will not be removed, which leaves it as an orphan.